x/stakeibc/keeper/msg_server_update_trade_route.go

```diff
@@ -35,15 +58,12 @@ func (ms msgServer) UpdateTradeRoute(goCtx context.Context, msg *types.MsgUpdate
 	updatedConfig := types.TradeConfig{
 		PoolId: msg.PoolId,
 
-	routes := k.GetAllTradeRoutes(ctx)
-	for _, route := range routes {
-		if route.HostDenomOnHostZone == msg.HostDenom && route.RewardDenomOnRewardZone == msg.RewardDenom {
-			route.PoolId = msg.PoolId
-			route.MinSwapAmount = msg.MinSwapAmount
-			route.MaxSwapAmount = msg.MaxSwapAmount
-			k.SetTradeRoute(ctx, route)
-		}
-		// if no matching trade route was found for the given host-denom and reward-denom... do nothing
+		SwapPrice:            sdk.ZeroDec(),
+		PriceUpdateTimestamp: 0,
+
+		MaxAllowedSwapLossRate: sdk.MustNewDecFromStr(maxAllowedSwapLossRate),
+		MinSwapAmount:          maxSwapAmount,
```

Comment:
Hold on both of these are wrong, the minSwapAmount needs to take the msg value but the max amount needs to take the computed maxAmount

Reply:
yikes, good catch!